test/functional/feature_rbf.py

```diff
@@ -623,5 +642,130 @@ def test_replacement_relay_fee(self):
+
+        # clean up all evicted utxos / update wallet utxo state
+        self.wallet.get_utxo(txid=tx['txid'])
+        self.wallet.get_utxo(txid=replacement_tx['txid'])
+
+    def test_reorged_inherited_signaling_and_descendant_limit(self):
+        confirmed_utxos = [self.wallet.get_utxo(), self.wallet.get_utxo()]
+
+        # One opt-in parent transaction, one opt-out
+        parent_txs = list(map(lambda utxo: self.wallet.send_self_transfer(
+            from_node=self.nodes[0],
+            utxo_to_spend=utxo,
+            sequence=BIP125_SEQUENCE_NUMBER if confirmed_utxos.index(utxo) == 0 else 0xffffffff,
+            fee_rate=Decimal('0.0001'),
+        ), confirmed_utxos))
+        optin_parent_tx, optout_parent_tx = parent_txs
+
+        # Craft a transaction that spends both parents so we can create one chain of descendant transactions
+        parent_utxos = list(map(lambda tx: self.wallet.get_utxo(txid=tx['txid']), parent_txs))
+        joined_tx_hex = self.create_double_input_self_transfer(parent_utxos, Decimal('0.0001'), sequence=0xffffffff)
+        joined_tx_txid = self.nodes[0].sendrawtransaction(joined_tx_hex)
+
+        # Only a single input needs to signal replaceability (whether explicitly or through inheritance) for the resulting transaction to signal
+        assert_equal(True, self.nodes[0].getmempoolentry(joined_tx_txid)['bip125-replaceable'])
+
+        # Get the `joined_tx` utxo into our wallet so we can spend a chain of descendants from it
+        joined_tx = self.nodes[0].decoderawtransaction(joined_tx_hex)
+        self.wallet.scan_tx(joined_tx)
+        joined_utxo = self.wallet.get_utxo(txid=joined_tx_txid)
+
+        # If we confirm `optin_parent_tx` descendants will no longer inherit signaling
+        hash = self.generateblock(self.nodes[0], output=self.wallet.get_address(), transactions=[optin_parent_tx['txid']])['hash']
+        assert_equal(False, self.nodes[0].getmempoolentry(joined_tx_txid)['bip125-replaceable'])
+
+        # Get `optin_parent_tx` back in our mempool, once again descendants inherit signaling
+        self.nodes[0].invalidateblock(hash)
+        assert_equal(True, self.nodes[0].getmempoolentry(joined_tx_txid)['bip125-replaceable'])
+
+        # Create a chain the size of `MAX_REPLACEMENT_LIMIT` spending `joined_tx` - the BIP125 Rule #5 imposed limit
+        tx = None
+        for _ in range(MAX_REPLACEMENT_LIMIT - 1):
+            tx = self.wallet.send_self_transfer(
+                from_node=self.nodes[0],
+                utxo_to_spend=joined_utxo if tx is None else self.wallet.get_utxo(txid=tx['txid']),  # a straight line of descendants
+                sequence=0xffffffff,
+                fee_rate=Decimal('0.0001'),
+            )
+            assert_equal(True, self.nodes[0].getmempoolentry(tx['txid'])['bip125-replaceable'])  # inherited
+        # Now we have a chain of: `optin_parent_tx`, `joined_tx`, and 99 txs. The last tx in the loop exceeded `MAX_REPLACEMENT_LIMIT`
```

Comment:
nit: My understanding is that it's better not to have actual values in comments
```suggestion
        # Now we have a chain of: `optin_parent_tx`, `joined_tx`, and `MAX_REPLACEMENT_LIMIT - 1` txs. The last tx in the loop exceeded `MAX_REPLACEMENT_LIMIT`
```